doc/Development_Documentation/23_Installation_and_Upgrade/03_System_Setup_and_Hosting/03_File_Storage_Setup.md

```diff
@@ -111,4 +159,63 @@ flysystem:
             options:
                 client: 'assets_s3'
                 bucket: 'bucket-name'
+                prefix: assets
 ```
+
+### Storage Migration
+If you are switching to different a storage type, it is often required to migrate contents from old storage to the newly configured one. Pimcore provides a command to solve the purpose of migrating contents from between storages, which in turn uses the flysystem `listcontents` API to read contents recursively from old (source) storage and copy contents to the new (destination) storage. 
+
+Follow these steps to migrate the content:
+1. Create a Flysystem configuration for source & destination storages. It is important to follow the naming structure, as follows: source config node: `pimcore.{storagetype}_source` & destination config node: `pimcore.{storagetype}_destination`
+
+Few examples,
+
+   | Migration Task | source node                        | destination node                      |
+   |----------------|------------------------------------|---------------------------------------|
+   | `asset`        | pimcore.asset_source.storage       | pimcore.asset_destination.storage     |
+   | `thumbnail`    | pimcore.thumbnail_source.storage   | pimcore.thumbnail_destination.storage |
+   | `version`      | pimcore.version_source.storage     | pimcore.version_source.storage        |
+   
+```yaml
+flysystem:
+    storages:
+        pimcore.asset_source.storage:
+            adapter: 'local'
+            visibility: public
+            options:
+                directory: '%kernel.project_dir%/public/var/assets'
+
+        pimcore.asset_destination.storage:
```

Comment:
```suggestion
        pimcore.asset.storage.destination:
```
@dvesh3 What do you think about this naming convention?

Reply:
@brusch Right now we are using `Storage::get()` which uses format `pimcore.%s.storage` to get the service https://github.com/pimcore/pimcore/blob/10.x/lib/Tool/Storage.php#L37.

Reply:
maybe `pimcore.asset.destination.storage`?

Reply:
Aaaah, I see. But we can just use the locator directly in the command, right? In my opinion a suffix would be just easier and looks a bit nicer.

Reply:
Updated the command to use locator directly and use suffix for storage names